polyply/src/load_library.py

```diff
@@ -13,75 +13,112 @@
 # limitations under the License.
 
 import os
-import pathlib
+from pathlib import Path
 import vermouth
 from vermouth.gmx.rtp import read_rtp
 from vermouth.citation_parser import read_bib
+from vermouth.log_helpers import StyleAdapter, get_logger
 from polyply import DATA_PATH
 from .ff_parser_sub import read_ff
+from .build_file_parser import read_build_file
 from .polyply_parser import read_polyply
 
+LOGGER = StyleAdapter(get_logger(__name__))
+BUILD_FILE_PARSERS = {'bld': read_build_file}
 FORCE_FIELD_PARSERS = {'rtp': read_rtp, 'ff': read_ff, 'itp': read_polyply, 'bib': read_bib}
 
+def get_parser(file_, file_parsers, is_lib_file):
+    """
+    check if file can be parsed and
+    if possible return the respective parser
+
+    Parameters
+    ----------
+    file_: class:`pathlib.PosixPath`
+        path to file
+    file_parsers: dict
+        dictionary of available file parsers
+    is_lib_file: bool
+        indicates whether the provided path is from a data library
+    """
+    file_extension = file_.suffix[1:]
+    if file_extension in file_parsers:
+        return file_parsers[file_extension]
+    elif not is_lib_file:
+        msg = "Cannot parse user provided file with extension {file_extension}."
+        raise IOError(msg)
 
-def _resolve_lib_paths(lib_names, data_path):
+
+def _resolve_lib_files(lib_names, data_path):
     """
-    select the appropiate files from data_path
+    select the appropiate files from a file path
     according to library names given.
 
     Parameters
     ----------
-    lib_names: list[`os.path`]
+    lib_names: list[`pathlib.Path`]
         list of library names
     data_path:
         location of library files
     """
     files = []
-    for name in lib_names:
-        directory = os.path.join(data_path, name)
-        for _file in os.listdir(directory):
-            _file = os.path.join(directory, _file)
-            files.append(_file)
+    data_path = Path(data_path)
+    for name in lib_names or []:
+        directory = data_path.joinpath(name)
+        for file_ in os.listdir(directory):
+            file_path = directory.joinpath(file_)
+            files.append(file_path)
     return files
 
 
-def read_ff_from_files(paths, force_field):
+def read_options_from_files(paths, storage_object, file_parsers):
     """
-    read the input files for the defintion of blocks and links.
+    read the input files for the definition of blocks and links.
 
     Parameters
     ----------
-    paths: list
-           List of vaild file paths
-    force_field: class:`vermouth.force_field.ForceField`
-
-    Returns
-    -------
-    force_field: class:`vermouth.force_field.ForceField`
-       updated forcefield
+    paths: list[`pathlib.Path`]
```

Comment:
based on line 93 paths needs to contain exactly 2 lists with lib_files and user_files. We should add some more information to the docstring about the requirements